version2/aks-cluster.tf

```diff
@@ -0,0 +1,64 @@
+resource "random_pet" "prefix" {}
+
+provider "azurerm" {
+  subscription_id = var.subscription_id
+  client_id       = var.client_id
+  client_secret   = var.client_secret
+  tenant_id       = var.tenant_id
+  version = "~> 2.0"
+  features {}
+}
+
+resource "azurerm_resource_group" "default" {
+  name     = "${random_pet.prefix.id}-rg"
+  location = "West US 2"
+
+  tags = {
+    environment = "Demo"
+  }
+}
+
+resource "azurerm_kubernetes_cluster" "default" {
+  name                = "${random_pet.prefix.id}-aks"
+  location            = azurerm_resource_group.default.location
+  resource_group_name = azurerm_resource_group.default.name
+  dns_prefix          = "${random_pet.prefix.id}-k8s"
+
+  default_node_pool {
+    name            = "default"
+    node_count      = 1
+    vm_size         = "Standard_D2_v2"
+    os_disk_size_gb = 30
+  }
+
+  linux_profile {
+      admin_username = "k8s"
+
+      ssh_key {
+          key_data = file(var.ssh_public_key)
+      }
+  }
```

Comment:
Planning on removing this too, this repo is meant to serve as a quick way to get started provisioning an AKS cluster

Users can then customize their cluster according to their use case by referencing the [`azurerm` Registry](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster) or referencing the [examples](https://github.com/terraform-providers/terraform-provider-azurerm/tree/master/examples/kubernetes) listed on the `terraform-provider-azurerm` GitHub page

I'll add links to the `terraform-provider-azurerm` GitHub page examples on the [Learn AKS guide](https://learn.hashicorp.com/tutorials/terraform/aks?in=terraform/kubernetes)